### ISSUE TITLE
Proposed: Allow users to encrypt EC2 Access and Secret keys in their configuration

### ISSUE THREAD

**Issue by andrewwatson** [docker/machine#634]
Some people (notably, me) aren't comfortable with sensitive information lurking in configuration files.  I'd like to make it possible for users to encrypt certain fields in their machine configurations such that they either have to specify a passphrase as a cli option or be prompted for it ever time they run a machine command.

**Comment by sthulb** [comment]
It's definitely something we need to do!

**Comment by nathanleclaire** [comment]
Yes.  This is an important issue for me and I ultimately want to move away from storing the secrets on a per-driver basis entirely, instead having one unified `Config` store where various configuration options including API keys are kept which has a few "backends" including one for plaintext as well as one for encryption.

**Comment by ehazlett** [comment]
+1 for encryption

**Comment by xiaohui** [comment]
+1
